File: src/main/java/io/javalin/ExceptionHandler.java
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin;

@FunctionalInterface
public interface ExceptionHandler<T extends Exception> {
    void handle(T exception, Request request, Response response);
}

File: src/main/kotlin/io.javalin/ExceptionHandler.kt
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin

@FunctionalInterface
interface ExceptionHandler<in T : Exception> {
    fun handle(exception: T, request: Request, response: Response)
}
